api_tests/scopes/views/test_scope_detail.py

```diff
@@ -0,0 +1,40 @@
+import pytest
+
+from api.base.settings.defaults import API_BASE
+from framework.auth.oauth_scopes import public_scopes
+
+
+@pytest.mark.django_db
+class TestScopeDetail:
+
+    def test_scope_detail(self, app):
+        count = 0
+        expected_count = len(public_scopes)
+        for key, value in public_scopes.iteritems():
+            id = key
+            description = value.description
+            is_public = value.is_public
+            url_scope = '/{}scopes/{}/'.format(API_BASE, id)
+
+            if is_public:
+                res_scope = app.get(url_scope)
+                data_scope = res_scope.json['data']
+                # test_scope_detail_success
+                assert res_scope.status_code == 200
+                assert res_scope.content_type == 'application/vnd.api+json'
+
+                # test_scope_top_level
+                assert data_scope['type'] == 'scopes'
+                assert data_scope['id'] == id
+
+                # test_scope_attributes:
+                assert data_scope['attributes']['description'] == description
+                assert data_scope['attributes']['description'] != ''
+                count += 1
+            else:
+                res_scope = app.get(url_scope, expect_errors=True)
+                assert 'data' not in res_scope.json
+                assert res_scope.status_code == 401
+                count += 1
+        assert count == expected_count
```

Comment:
This assertion seems suspect. `count` is incremented in both the `if` and the `else`, so this is just checking that the number of keys in `public_scopes` is equal to the number of items in `public_scopes`, which will always be `True`.

Reply:
Should be. The only reason for the assert is to make sure that every item in the list is tested versus an assertion and we don't skip over some for some reason, as there are no other assertions outside of an if block. With this simple test, it should be fine, but if something more complicated were put in later to limit our usage of scopes, this type of mechanism ensures that we don't just, for example, not test anything at all.

Reply:
Fine with me!